### PR TITLE
SUBMARINE-261. Add exception class in module common utils.

### PR DESCRIPTION
**Pull request by yuanzac**
### What is this PR for?
Add several exception classes for submarine modules.

### What type of PR is it?
Improvement

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-261

### How should this be tested?
https://travis-ci.org/yuanzac/hadoop-submarine/builds/602727483

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-client/src/main/java/org/apache/submarine/client/cli/CliUtils.java [hunk 1]
```diff
@@ -89,24 +89,24 @@ public static void doLoginIfSecure(String keytab, String principal) throws
 
     if (StringUtils.isEmpty(keytab) || StringUtils.isEmpty(principal)) {
       if (StringUtils.isNotEmpty(keytab)) {
-        SubmarineRuntimeException e = new SubmarineRuntimeException("The " +
-            "parameter of " + CliConstants.PRINCIPAL + " is missing.");
+        SubmarineCliRuntimeException e = new SubmarineCliRuntimeException(
+            "The parameter of " + CliConstants.PRINCIPAL + " is missing.");
```
Comment (adamantal) [inline submarine-client/src/main/java/org/apache/submarine/client/cli/CliUtils.java]:
```suggestion
            String.format("The parameter of %s is missing.", CliConstants.PRINCIPAL));
```

File: submarine-client/src/main/java/org/apache/submarine/client/cli/CliUtils.java [hunk 2]
```diff
@@ -89,24 +89,24 @@ public static void doLoginIfSecure(String keytab, String principal) throws
...
         LOG.error(e.getMessage(), e);
         throw e;
       }
 
       if (StringUtils.isNotEmpty(principal)) {
-        SubmarineRuntimeException e = new SubmarineRuntimeException("The " +
-            "parameter of " + CliConstants.KEYTAB + " is missing.");
+        SubmarineCliRuntimeException e = new SubmarineCliRuntimeException(
+            "The parameter of " + CliConstants.KEYTAB + " is missing.");
```
Comment (adamantal) [inline submarine-client/src/main/java/org/apache/submarine/client/cli/CliUtils.java]:
```suggestion
            String.format("The parameter of %s is missing.", CliConstants.KEYTAB));
```

File: submarine-client/src/main/java/org/apache/submarine/client/cli/CliUtils.java [hunk 3]
```diff
@@ -89,24 +89,24 @@ public static void doLoginIfSecure(String keytab, String principal) throws
...
       }
 
       UserGroupInformation user = UserGroupInformation.getCurrentUser();
       if (user == null || user.getAuthenticationMethod() ==
           UserGroupInformation.AuthenticationMethod.SIMPLE) {
-        SubmarineRuntimeException e = new SubmarineRuntimeException("Failed " +
-            "to authenticate in secure environment. Please run kinit " +
+        SubmarineCliRuntimeException e = new SubmarineCliRuntimeException(
+            "Failed to authenticate in secure environment. Please run kinit " +
             "command in advance or use " + "--" + CliConstants.KEYTAB + "/--" + CliConstants.PRINCIPAL +
```
Comment (adamantal) [inline submarine-client/src/main/java/org/apache/submarine/client/cli/CliUtils.java]:
```suggestion
            "command in advance or use " + "--" + CliConstants.KEYTAB + " or --" + CliConstants.PRINCIPAL +
```

File: submarine-commons/commons-utils/src/main/java/org/apache/submarine/commons/utils/exception/SubmarineCliRuntimeException.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.submarine.commons.utils.exception;
+
+public class SubmarineCliRuntimeException extends SubmarineRuntimeException {
```
Comment (adamantal) [inline submarine-commons/commons-utils/src/main/java/org/apache/submarine/commons/utils/exception/SubmarineCliRuntimeException.java]: Could you please add a javadoc to this? E.g. it's Submarine's CLI-related RuntimeExceptions.

File: submarine-commons/commons-utils/src/main/java/org/apache/submarine/commons/utils/exception/SubmarineInterpreterException.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.submarine.commons.utils.exception;
+
+public class SubmarineInterpreterException extends SubmarineException {
```
Comment (adamantal) [inline submarine-commons/commons-utils/src/main/java/org/apache/submarine/commons/utils/exception/SubmarineInterpreterException.java]: Could you please add a javadoc to this?

**Comment by jiwq** [comment]
@yuanzac Are you still working for this? If not maybe I think we can close this pr. Any thoughts?

**Pull request closed without merging**